packages/components/src/functional-components/Heading/Heading.tsx

```diff
@@ -0,0 +1,77 @@
+import { h, type FunctionalComponent as FC } from '@stencil/core';
+import clsx from 'clsx';
+import type { JSXBase, VNode } from '@stencil/core/internal';
+import type { HeadingLevel, HeadingVariantPropType } from '../../schema';
+
+type LevelProps = { level?: HeadingLevel };
+type VariantProps = { variant?: HeadingVariantPropType };
+
+type BaseProps = JSXBase.HTMLAttributes<HTMLHeadingElement> & LevelProps;
+type HeadlineProps = BaseProps & VariantProps;
+type SecondaryHeadlineProps = BaseProps;
+
+export type HeadingProps = BaseProps &
+	VariantProps & {
+		secondaryHeadline?: string;
+	};
+
+function getHeadlineTag(level: HeadingLevel | number): string {
+	const validHeadline = level > 0 && level <= 6;
+	return validHeadline ? `h${level}` : 'strong';
+}
+
+function getSubHeadlineTag(level: HeadingLevel | number): string {
+	if (level === 1) {
+		return 'span';
+	}
+
+	return getHeadlineTag(level);
+}
+
+const KolHeadlineFc: FC<HeadlineProps> = (props, children) => {
+	const { class: classNames, level = 1, variant: defaultVariant, ...other } = props;
+
+	const HeadlineTag = getHeadlineTag(level);
+	const variant = defaultVariant || HeadlineTag;
+
+	return (
+		<HeadlineTag class={clsx('headline', `headline-${variant}`, classNames)} {...other}>
+			{children}
+		</HeadlineTag>
+	);
+};
+
+const KolSecondaryHeadlineFc: FC<SecondaryHeadlineProps> = (props, children) => {
+	const { class: classNames, level = 1, ...other } = props;
+
+	const HeadlineTag = getSubHeadlineTag(level + 1);
+
+	return (
+		<HeadlineTag class={clsx('secondary-headline', classNames)} {...other}>
+			{children}
+		</HeadlineTag>
+	);
+};
+
+const KolHeadingFc: FC<HeadingProps> = (props, children) => {
+	const { ref, secondaryHeadline, level = 1, class: classNames, ...other } = props;
+
+	let result: unknown = (
+		<KolHeadlineFc ref={ref} level={level} {...other}>
+			{children}
+		</KolHeadlineFc>
+	);
+
+	if (secondaryHeadline) {
+		result = (
+			<hgroup>
+				{result}
+				<KolSecondaryHeadlineFc level={level}>{secondaryHeadline}</KolSecondaryHeadlineFc>
+			</hgroup>
+		);
+	}
+
+	return <section class={clsx('kol-heading-wc', classNames)}>{result as VNode}</section>;
```

Comment:
Hi @Makko74, @sdvg, @cbraehmig,

meines Erachtens ist `section` hier falsch verwendet.

Die `<section>`-Elemente in HTML dienen dazu, **inhaltliche Abschnitte einer Webseite zu gliedern**. Eine `<section>` wird verwendet, um thematisch zusammengehörige Inhalte zu gruppieren, die oft eine eigene Überschrift oder Struktur haben. Dadurch wird die semantische Bedeutung des Inhalts hervorgehoben und die Struktur der Webseite klarer und besser verständlich – sowohl für Menschen als auch für Suchmaschinen und andere Werkzeuge (z. B. Screenreader).

### Typische Verwendung von `<section>`:
1. **Themenbezogene Inhalte**: Abschnitte zu einem bestimmten Thema, die in sich geschlossen sind.
2. **Mit einer Überschrift**: Jede `<section>` sollte idealerweise eine eigene Überschrift haben (z. B. `<h1>` bis `<h6>`).
3. **Dokumentstruktur**: Um die Hierarchie der Inhalte auf einer Seite zu verdeutlichen.

### Beispiel:
```html
<article>
  <section>
    <h2>Einleitung</h2>
    <p>Dies ist die Einführung in das Thema.</p>
  </section>
  <section>
    <h2>Hauptteil</h2>
    <p>Hier steht der Hauptinhalt.</p>
  </section>
  <section>
    <h2>Fazit</h2>
    <p>Das ist die Schlussfolgerung des Artikels.</p>
  </section>
</article>
```

### Vorteile:
1. **Semantische Struktur**: Klare Bedeutung der Inhalte durch die Einteilung in Abschnitte.
2. **SEO (Suchmaschinenoptimierung)**: Inhalte werden besser von Suchmaschinen indiziert.
3. **Barrierefreiheit**: Screenreader können die Struktur der Seite leichter interpretieren.
4. **Einfache Navigation**: Ermöglicht es, sich durch große Mengen an Inhalt zu navigieren.

### Wann sollte `<section>` verwendet werden?
- Wenn der Inhalt thematisch getrennt ist und für sich stehen könnte.
- Wenn es sinnvoll ist, den Abschnitt mit einer Überschrift zu versehen.

### Unterschied zu anderen HTML-Tags:
- **`<div>`**: Wird für generische Container verwendet und hat keine semantische Bedeutung.
- **`<article>`**: Wird für eigenständige Inhalte verwendet, die unabhängig von der Seite existieren könnten (z. B. Blog-Post, News-Artikel).
- **`<aside>`**: Für Inhalte, die ergänzend oder nicht im Hauptfokus stehen (z. B. Seitenleiste).

Mit `<section>` wird die Struktur der Webseite besser lesbar, logisch gegliedert und semantisch korrekt.

Reply:
Dieser Wrapper ist komplertt entfernt